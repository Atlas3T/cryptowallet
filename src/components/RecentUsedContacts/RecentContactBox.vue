<template>
  <div class="contactBox">
    <router-link
      id="btn_id"
      tag="button"
      to="/#FixThePath">
      <div class="item">
        <img src="~/assets/avatar.png">
        <p id="usersName_id">Name Surname</p>
      </div>&nbsp;
    </router-link>
  </div>
</template>

<script>

export default {

  // data() {
  //   return {

  //   };
  // },

  // beforeCreate() {

  // },

  // created() {

  // },

  // beforeMount() {

  // },

  // mounted() {

  // },

  // beforeUpdate() {

  // },

  // updated() {

  // },

  // beforeDestroy() {

  // },

  // destroyed() {

  // },

  // methods: {

  // },

};


</script>

<style lang="scss" scoped>

  .item {
    text-align: center;
    position: sticky;
    display: inline-block;
    width: 150px;
    height: 140px;
    border-radius: 50px;
    /* border: 3px solid rgb(41, 128, 187); */
  }

  img {
    height: 70px;
    width: 45%;
  }

  p {
    #usersName_id {
      color: white;
    }
  }

  #btn_id {
    background-color: Transparent;
    border: none;
    cursor:pointer;
    overflow: hidden;
    outline:none;
  }

</style>
