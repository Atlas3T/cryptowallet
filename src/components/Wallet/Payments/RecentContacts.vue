<template>
  <section
    v-show="!isSearchingContacts"
    class="recent-contacts-list"
  >
    <h1 class="section-h1">{{ $t('recent') }}</h1>
    <div class="flex-scroll-area-wrapper">
      <div class="test">
        <q-scroll-area class="recent-contacts-scroll-area">
          <div class="scroll-content">
            <RecentContactItem
              v-for="contact in recentContacts"
              :key="contact.name"
              :contact="contact"
            />
          </div>
        </q-scroll-area>
      </div>
    </div>
  </section>
</template>

<script>
import { mapState } from 'vuex';
import RecentContactItem from './RecentContactItem.vue';

export default {
  name: 'RecentContacts',
  components: {
    RecentContactItem,
  },
  computed: {
    ...mapState({
      isSearchingContacts: (state) => { return state.payments.isSearchingContacts; },
      recentContacts: (state) => { return state.wallet.recentContacts; },
    }),
  },
};
</script>

<style scoped>
.test {
    position: absolute;
    width: 100%;
    height: 100%;
}

.section-h1 {
  color: white;
  font-size: 0.9rem;
  font-family: 'Montserrat-Light';
  background: #162f46;
  padding: 0.5rem 1rem;
  border-bottom: 1px solid #0a2338;
  letter-spacing: 0.05rem;
  text-transform: uppercase;
}

.flex-scroll-area-wrapper {
  flex: auto;
  flex-grow: 0;
  position: relative;
  height: 100%;
}

.recent-contacts-scroll-area {
  display: flex;
  height: 100%;
}

.recent-contacts-list {
  height: 25vh;
  border-top: 2px solid #09233a;
  position: relative;
  background: #16324a;
  display: flex;
  flex-direction: column;
  transition: height 100ms ease-in-out;
}

.scroll-content {
  padding: 0 1rem;
  display: flex;
  height: 100%;
  align-items: center;
}
</style>
