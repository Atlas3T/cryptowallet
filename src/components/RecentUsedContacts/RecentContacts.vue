<template>
  <div class="mainContentContacts">
    <p> &nbsp; </p>
    <p id="Title_id">
      &nbsp;{{ $t("recent") }}
    </p>
    <hr>
    <div class="container is-fluid">
      <div id="items">
&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;
        <div class="item">
          <RecentContactBox />
        </div>&nbsp;&nbsp;
        <br>
      </div>
    </div>
    <br>
  </div>
</template>

<script>
import RecentContactBox from './RecentContactBox.vue';

export default {

  components: {
    RecentContactBox,
  },

  data() {
    return {

    };
  },

  beforeCreate() {

  },

  created() {

  },

  beforeMount() {

  },

  mounted() {

  },

  beforeUpdate() {

  },

  updated() {

  },

  beforeDestroy() {

  },

  destroyed() {

  },

  methods: {

  },

};


</script>

<style scoped>

.mainContentContacts {
  background-color:transparent;
  width: auto;
  height: auto;
  margin: auto;
}

#Title_id {
  color: white;
  font-size: 18px;
  font-weight: bold;
  margin-left: 10px;
  }

  #items{
  overflow-x: scroll;
  width: auto;
  white-space: nowrap;
  }
  .item{
    text-align: center;
    position: sticky;
    display: inline-block;
  }

  img {
  height: 70px;
  width: 45%;
}

</style>
