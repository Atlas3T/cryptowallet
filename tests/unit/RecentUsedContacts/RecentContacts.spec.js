import { shallowMount } from '@vue/test-utils';
import RecentContacts from '@/components/RecentUsedContacts/RecentContacts.vue';
import RecentContactBox from '@/components/RecentUsedContacts/RecentContactBox.vue';
import { localVue } from '../setupLocalVue';

describe('RecentContacts.vue', () => {
  let wrapper;

  function wrapperInit(options) {
    return shallowMount(RecentContacts, options);
  }

  beforeEach(() => {
    wrapper = wrapperInit({ localVue });
  });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });


  it('expect main content', () => {
    expect(wrapper.contains('.mainContentContacts')).toBe(true);
  });

  it('expect title', () => {
    expect(wrapper.contains('#Title_id')).toBe(true);
  });

  it('expect container', () => {
    expect(wrapper.contains('div.container')).toBe(true);
  });

  it('expect items', () => {
    expect(wrapper.contains('div#items')).toBe(true);
  });

  it('expect item', () => {
    expect(wrapper.contains('div.item')).toBe(true);
  });

  it('expect import contact Box', () => {
    expect(wrapper.contains(RecentContactBox)).toBe(true);
  });
});
