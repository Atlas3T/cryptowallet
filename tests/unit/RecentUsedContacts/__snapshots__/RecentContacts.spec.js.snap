// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentContacts.vue renders and matches snapshot 1`] = `
<div
  class="mainContentContacts"
>
  <p
    id="Title_id"
  >
     Recent
  </p>
   
  <hr />
   
  <div
    class="container is-fluid"
  >
    <div
      id="items"
    >
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
       
      
      <div
        class="item"
      >
        <recentcontactbox-stub />
      </div>
        
      
      <br />
    </div>
  </div>
   
  <br />
</div>
`;
